Sources/Utility/Process.swift

```diff
@@ -0,0 +1,384 @@
+/*
+ This source file is part of the Swift.org open source project
+ 
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ 
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import class Foundation.ProcessInfo
+
+import enum POSIX.SystemError
+import libc
+import Basic
+
+/// The POSIX signals.
+// FIXME: Move this to POSIX layer.
+public enum Signal: Int32 {
+    /// Interrupt signal.
+    case interrupt
+
+    /// Kill signal.
+    case kill
+
+    init?(_ rawValue: Int32) {
+        switch rawValue {
+        case SIGINT: self = .interrupt
+        case SIGKILL: self = .kill
+        default: return nil
+        }
+    }
+
+    public var rawValue: Int32 {
+        switch self {
+        case .interrupt: return SIGINT
+        case .kill: return SIGKILL
+        }
+    }
+}
+
+/// Process result data which is available after process termination.
+public struct ProcessResult {
+    public enum ExitStatus {
+        /// The process was terminated normally with a exit code.
+        case terminated(code: Int32)
+
+        /// The process was terminated due to a signal.
+        case signalled(signal: Int32)
+    }
+
+    /// The exit status of the process.
+    public let exitStatus: ExitStatus
+
+    /// The output of the process. Available only if the process was asked to redirect its output.
+    public let output: Result<String, AnyError>
+
+    /// Create an instance using the process exit code and output result.
+    fileprivate init(exitCode: Int32, output: Result<String, AnyError>) {
```

Comment:
Technically, `exitCode` should be `exitStatus` here -- the code is the value returned from a non-signaled process.